src/components/modules/dragNDrop.ts

```diff
@@ -89,28 +113,104 @@ export default class DragNDrop extends Module {
      * Try to set current block by drop target.
      * If drop target is not part of the Block, set last Block as current.
      */
-    const targetBlock = BlockManager.setCurrentBlockByChildNode(dropEvent.target as Node);
+    let targetBlock = BlockManager.setCurrentBlockByChildNode(dropEvent.target as Node);
 
     if (targetBlock) {
-      this.Editor.Caret.setToBlock(targetBlock, Caret.positions.END);
+      if (targetBlock.dropTarget === BlockDropZonePosition.Top) {
+        const currentIndex = BlockManager.getBlockIndex(targetBlock);
+
+        if (currentIndex > 0) {
+          targetBlock = BlockManager.getBlockByIndex(currentIndex - 1);
+          Caret.setToBlock(targetBlock, Caret.positions.END);
+        }
+        /**
+         *  @todo -Implement to drop a block before the first block,
```

Comment:
Will you do it in this PR or in a separate one?

Reply:
yup

Reply:
yup

Reply:
Soo 😄